core/node/consensus/src/tests.rs

```diff
@@ -616,8 +616,16 @@ async fn test_with_pruning(version: ProtocolVersionId) {
             .wait_for_batch(ctx, validator.last_sealed_batch())
             .await?;
 
+        // The main node is not supposed to be pruned. In particular `ConsensusDal::attestation_status`
+        // does not look for where the last prune happened at, and thus if we prune the block genesis
+        // points at, we might never be able to start the Executor.
```

Comment:
the main node is not pruned currently, but there is nothing preventing us from doing that. FWIW, we actually plan to prune it at some point. How hard is it to not rely on this? No need to do now, but let's at least create a ticket to resolve this

Reply:
Not committed batches are never pruned. We don't have such nice guarantee here, because we only simulate the L1 interaction, but once committing a batch requires attestation certificate, pruning won't be an issue. There is nothing to fix there.

Reply:
For the record the very first attempt that looked at pruning was [here](https://github.com/matter-labs/zksync-era/pull/2480/commits/3953ffd838dc644371aebbf7f860809a4f33ecf1#diff-9958fba2c9e6d7938fdd3bea6e06596e3cff47eb06dda4e440f9117f342495d0R51).